<app-ncats-header [searchBar]="true"></app-ncats-header>
<div fxLayout="row" fxLayout.lt-sm="column" class="about-content">
  <div fxFlex="20" fxFlex.lt-sm="90" class="toc-wrapper">
    <div class="toc">
      <mat-nav-list>
        <a mat-list-item class="list-button" (click)="scroll(introduction)" [ngClass]="{'active': isActive('introduction')}">
          Introduction
        </a>
        <a mat-list-item class="list-button" (click)="scroll(collaborators)"
                       [ngClass]="{'active': isActive('collaborators')}">Our Collaborators
        </a>
        <a mat-list-item class="list-button" (click)="scroll(availabledata)"
                       [ngClass]="{'active': isActive('availabledata')}">Available Data
        </a>
        <a mat-list-item class="list-button" (click)="scroll(datatypes)" [ngClass]="{'active': isActive('datatypes')}">Data
          Types
        </a>
        <a mat-list-item class="list-button" (click)="scroll(datasources)" [ngClass]="{'active': isActive('datasources')}">Data
          Sources
        </a>
        <a mat-list-item class="list-button" (click)="scroll(datadownload)" [ngClass]="{'active': isActive('datadownload')}">
          Data Download
        </a>
        <a mat-list-item class="list-button" (click)="scroll(license)" [ngClass]="{'active': isActive('license')}">Pharos License
        </a>
        <a mat-list-item class="list-button" (click)="scroll(code)" [ngClass]="{'active': isActive('code')}">Pharos Code
        </a>
        <a mat-list-item class="list-button" (click)="scroll(attribution)" [ngClass]="{'active': isActive('attribution')}">
          Attribution
        </a>
        <a mat-list-item class="list-button" (click)="scroll(help)" [ngClass]="{'active': isActive('help')}">
          Help
        </a>
      </mat-nav-list>
    </div>
  </div>
  <div fxFlex="75" fxFlex.lt-sm="90" id="scroller-wrapper" class="about-wrapper" cdkScrollable>
    <div fxLayout="row" id = "introduction" #introduction #scrollSection>
      <div fxFlex="75">
        <div class="section-header" [ngClass]="{'active': isActive('introduction')}">Introduction</div>
        <div class="section-content">
          <p>Pharos is the user interface to the <a href="http://targetcentral.ws/">Knowledge Management Center</a>
            (KMC) for the <a href="https://commonfund.nih.gov/idg/index">Illuminating the Druggable Genome</a>
            (IDG) program funded by the National Institutes of Health (NIH) Common Fund.
            (Grant No. <a href="https://projectreporter.nih.gov/project_info_description.cfm?aid=9461406&icde=38627613">
              1U24CA224370-01</a>). The goal of KMC is to develop a comprehensive, integrated knowledge-base
            for the Druggable Genome (DG) to illuminate the uncharacterized and/or poorly annotated portion of
            the DG, focusing on four of the most commonly drug-targeted protein families: &nbsp;G-protein-coupled
            receptors (GPCRs); nuclear receptors (NRs); ion channels (ICs); and kinases.
          </p>
          <p>The Pharos interface provides facile access to all data types
            collected by the KMC.
            Given the complexity of the data surrounding any target, efficient and intuitive visualization has
            been a high priority, to enable users to quickly navigate &amp; summarize search results and
            rapidly identify patterns. A critical feature of the interface is the ability to perform flexible
            search and subsequent drill down of search results. Underlying the interface is a RESTful API that
            provides programmatic access to all KMC data, allowing for easy consumption in user applications.
          </p>
        </div>
      </div>
    </div>
    <div fxLayout="row" id ="collaborators" #collaborators #scrollSection>
      <div fxFlex="75">
        <div class="section-header" [ngClass]="{'active': isActive('collaborators')}">Our Collaborators</div>
        <div class="section-content">
          <p>Pharos is developed at <a href="http://ncats.nih.gov">NCATS</a>, together with collaborators from
            the <a href="http://hsc.unm.edu/">University of New Mexico</a>, <a href="http://icahn.mssm.edu/">
              Icahn School of Medicine, Mount Sinai</a>, <a href="https://www.ebi.ac.uk/about">EMBL-EBI</a>,
            the <a href="http://www.cpr.ku.dk/">
              Novo Nordisk Foundation Center for Protein Research (U. Copenhagen)</a>
            and the <a href="http://ccs.miami.edu/?p=1103">University of Miami</a>.
          </p>
          <p>
            The IDG TechDev network is the experimental side of IDG. TechDev research aims to develop and
            utilize technologies and streamlined experimental workflows for functional studies of poorly
            characterized and/or un-annotated genes and proteins, thereby providing high value data and knowledge
            for KMC. KMC in turn is tasked with providing guidance as to research priorities based on knowledge
            gaps and druggability likelihood analyses.
          </p>
          <p>
            To find more details on our collaborators visit the <a href="https://druggablegenome.net">
            IDG KMC</a>
          </p>
        </div>
      </div>
    </div>
    <div fxLayout="row" id="availabledata" #availabledata #scrollSection>
      <div fxFlex="75">
        <div class="section-header" [ngClass]="{'active': isActive('availabledata')}">Available Data</div>
        <div class="section-content">
          <p>The data available in Pharos is obtained from the <a href="http://juniper.health.unm.edu/tcrd/">
            Target Central Resource Database</a> which integrates data from a variey of data sources including
            the <a href="http://amp.pharm.mssm.edu/Harmonizome/">Harmonizome</a>, Jensen Lab datasets, EBI data
            sets (such as <a href="https://www.ebi.ac.uk/chembl/">ChEMBL</a>) and the
            <a href="http://drugtargetontology.org/about.html">Drug Target Ontology</a> (DTO)
            from U. Miami. TCRD integration methodology involves
            importation by value or by reference to external sources, as informed by performance, provenance,
            and other design criteria. See <a href="http://targetcentral.ws/Pharos">here</a> for a full listing
            of datasets incorporated in to <a href="http://juniper.health.unm.edu/tcrd/">TCRD</a>.
          </p>
        </div>
      </div>
    </div>
    <div fxLayout="row" id="datatypes" #datatypes #scrollSection>
      <div fxFlex="75">
        <div class="section-header" [ngClass]="{'active': isActive('datatypes')}">Data Types</div>
        <div class="section-content">
          <p>
            The key data types represented in Pharos are listed below:
          </p>
          <mat-list class="data-types-list">
            <mat-list-item>
              <div><span class="data-type">Small molecule data</span> including approved drug data, bioassay data</div>
            </mat-list-item>
            <mat-list-item>
              <div><span class="data-type">Protein data</span> including protein-protein interaction data</div>
            </mat-list-item>
            <mat-list-item>
              <div><span class="data-type">Disease data</span> from <a href="http://www.ncbi.nlm.nih.gov/omim">OMIM</a>
                and
                <a href="http://disease-ontology.org/">Disease Ontology</a></div>
            </mat-list-item>
            <mat-list-item>
              <div><span class="data-type">Genomic data</span> including expression (protein, RNA), transcription
                factors and epigenomic
              </div>
              associations
            </mat-list-item>
            <mat-list-item>
              <div><span class="data-type">Phenotypic data</span> including mouse phenotypes, mouse/human orthologs and
                GWAS results
              </div>
            </mat-list-item>
            <!--          <mat-list-item>
                        <div><span class="data-type">Funding data</span> via NIH <a href="https://projectreporter.nih.gov/reporter.cfm">RePORTER</a></div>
                      </mat-list-item> -->
            <mat-list-item>
              <div><span class="data-type">Text data</span> including <a
                href="http://www.ncbi.nlm.nih.gov/gene/about-generif">GeneRIF</a>'s
                and text-mined publications
              </div>
            </mat-list-item>
            <mat-list-item>
              <div><span class="data-type">Ontologies</span> including the Drug Target Ontology, <a
                href="http://disease-ontology.org/">Disease
                Ontology</a>, <a href="http://www.pantherdb.org/panther/ontologies.jsp">PANTHER</a> and
                <a href="http://geneontology.org/">GO</a></div>
            </mat-list-item>
          </mat-list>
        </div>
      </div>
    </div>
    <div fxLayout="row" id = "datasources" #datasources #scrollSection>
      <div fxFlex="75">
        <div class="section-header" [ngClass]="{'active': isActive('datasources')}">Data Sources</div>
        <div class="section-content">
        <p>
          The goal of the IDG KMC is to integrate a variety of data sources to shed light on unstudied and understudied
          targets. To achieve this we have pulled together data on protein targets, small molecule activity, genomic
          behavior and disease links. <!--For a list of datasources, and their associated entity counts, see the About
          Pharos page. A visual representation of all targets and their representation in all IDG datasources can be
          viewed here.--> Over time we anticipate incorporating other relevant data sources. </p>

        <table class="table table-striped">
          <tbody><tr>
            <th>Source</th>
            <th>Target</th>
            <th>Disease</th>
            <th>Ligand</th>
          </tr>

          <tr>


            <td>BioPlex</td>


            <td><a href="/idg/targets?facet=Data+Source/BioPlex">6400</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/BioPlex"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/BioPlex"></a>
            </td>
          </tr>

          <tr>

            <td><a href="https://www.ebi.ac.uk/chembl">CHEMBL</a></td>

            <td><a href="/idg/targets?facet=Data+Source/CHEMBL"></a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/CHEMBL"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/CHEMBL">221848</a>
            </td>
          </tr>

          <tr>

            <td><a href="https://ctdbase.org/">CTD</a></td>

            <td><a href="/idg/targets?facet=Data+Source/CTD">7446</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/CTD">5699</a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/CTD"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://wlab.ethz.ch/cspa/">Cell Surface Protein Atlas</a></td>

            <td><a href="/idg/targets?facet=Data+Source/Cell+Surface+Protein+Atlas">1038</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/Cell+Surface+Protein+Atlas"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/Cell+Surface+Protein+Atlas"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://targetcentral.ws">Consensus</a></td>

            <td><a href="/idg/targets?facet=Data+Source/Consensus">18503</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/Consensus"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/Consensus"></a>
            </td>
          </tr>

          <tr>


            <td><a href="https://dailymed.nlm.nih.gov/dailymed/">DRUG LABEL</a></td>


            <td><a href="/idg/targets?facet=Data+Source/DRUG+LABEL"></a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/DRUG+LABEL"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/DRUG+LABEL">219</a>
            </td>
          </tr>

          <tr>

            <td><a href="http://www.disgenet.org">DisGeNET</a></td>

            <td><a href="/idg/targets?facet=Data+Source/DisGeNET">7631</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/DisGeNET">12446</a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/DisGeNET"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://www.disease-ontology.org">DiseaseOntology</a></td>

            <td><a href="/idg/targets?facet=Data+Source/DiseaseOntology"></a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/DiseaseOntology"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/DiseaseOntology"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://drugcentral.org/">DrugCentral Indication</a></td>

            <td><a href="/idg/targets?facet=Data+Source/DrugCentral+Indication">911</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/DrugCentral+Indication">1234</a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/DrugCentral+Indication"></a>
            </td>
          </tr>

          <tr>


            <td><a href="http://eggnogdb.embl.de">EggNOG</a></td>


            <td><a href="/idg/targets?facet=Data+Source/EggNOG">16120</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/EggNOG"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/EggNOG"></a>
            </td>
          </tr>

          <tr>

            <td><a href="https://www.ebi.ac.uk/gxa/">Expression Atlas</a></td>

            <td><a href="/idg/targets?facet=Data+Source/Expression+Atlas">16585</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/Expression+Atlas">142</a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/Expression+Atlas"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://www.gtexportal.org/">GTEx</a></td>

            <td><a href="/idg/targets?facet=Data+Source/GTEx">18679</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/GTEx"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/GTEx"></a>
            </td>
          </tr>

          <tr>

            <td><a href="https://www.genome.gov/26525384">GWAS Catalog</a></td>

            <td><a href="/idg/targets?facet=Data+Source/GWAS+Catalog">10255</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/GWAS+Catalog"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/GWAS+Catalog"></a>
            </td>
          </tr>

          <tr>

            <td><a href="https://www.humancellatlas.org">HCA RNA</a></td>

            <td><a href="/idg/targets?facet=Data+Source/HCA+RNA">18868</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/HCA+RNA"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/HCA+RNA"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://tissues.jensenlab.org">HPA Protein</a></td>

            <td><a href="/idg/targets?facet=Data+Source/HPA+Protein">10414</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/HPA+Protein"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/HPA+Protein"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://tissues.jensenlab.org">HPA RNA</a></td>

            <td><a href="/idg/targets?facet=Data+Source/HPA+RNA">18499</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/HPA+RNA"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/HPA+RNA"></a>
            </td>
          </tr>

          <tr>

            <td><a href="">HPM Gene</a></td>

            <td><a href="/idg/targets?facet=Data+Source/HPM+Gene">16026</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/HPM+Gene"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/HPM+Gene"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://www.humanproteomemap.org">HPM Protein</a></td>

            <td><a href="/idg/targets?facet=Data+Source/HPM+Protein">16251</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/HPM+Protein"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/HPM+Protein"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://www.mousephenotype.org/data/secondaryproject/idg">IMPC</a></td>

            <td><a href="/idg/targets?facet=Data+Source/IMPC">3485</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/IMPC"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/IMPC"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://www.guidetopharmacology.org/">IUPHAR</a></td>

            <td><a href="/idg/targets?facet=Data+Source/IUPHAR"></a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/IUPHAR"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/IUPHAR">121</a>
            </td>
          </tr>

          <tr>


            <td><a href="http://inparanoid.sbc.su.se/">Inparanoid</a></td>


            <td><a href="/idg/targets?facet=Data+Source/Inparanoid">4903</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/Inparanoid"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/Inparanoid"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://www.informatics.jax.org/">JAX/MGI Human Ortholog Phenotype</a></td>

            <td><a href="/idg/targets?facet=Data+Source/JAX%2FMGI+Human+Ortholog+Phenotype">9720</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/JAX%2FMGI+Human+Ortholog+Phenotype"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/JAX%2FMGI+Human+Ortholog+Phenotype"></a>
            </td>
          </tr>

          <tr>


            <td>JensenLab Experiment COSMIC</td>


            <td><a href="/idg/targets?facet=Data+Source/JensenLab+Experiment+COSMIC">11445</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/JensenLab+Experiment+COSMIC">83</a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/JensenLab+Experiment+COSMIC"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://distild.jensenlab.org">JensenLab Experiment DistiLD</a></td>

            <td><a href="/idg/targets?facet=Data+Source/JensenLab+Experiment+DistiLD">3914</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/JensenLab+Experiment+DistiLD">209</a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/JensenLab+Experiment+DistiLD"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://tissues.jensenlab.org">JensenLab Experiment Exon array</a></td>

            <td><a href="/idg/targets?facet=Data+Source/JensenLab+Experiment+Exon+array">13611</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/JensenLab+Experiment+Exon+array"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/JensenLab+Experiment+Exon+array"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://tissues.jensenlab.org">JensenLab Experiment GNF</a></td>

            <td><a href="/idg/targets?facet=Data+Source/JensenLab+Experiment+GNF">10637</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/JensenLab+Experiment+GNF"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/JensenLab+Experiment+GNF"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://tissues.jensenlab.org">JensenLab Experiment HPA</a></td>

            <td><a href="/idg/targets?facet=Data+Source/JensenLab+Experiment+HPA">15665</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/JensenLab+Experiment+HPA"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/JensenLab+Experiment+HPA"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://tissues.jensenlab.org">JensenLab Experiment HPA-RNA</a></td>

            <td><a href="/idg/targets?facet=Data+Source/JensenLab+Experiment+HPA-RNA">16979</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/JensenLab+Experiment+HPA-RNA"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/JensenLab+Experiment+HPA-RNA"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://tissues.jensenlab.org">JensenLab Experiment HPM</a></td>

            <td><a href="/idg/targets?facet=Data+Source/JensenLab+Experiment+HPM">15886</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/JensenLab+Experiment+HPM"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/JensenLab+Experiment+HPM"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://tissues.jensenlab.org">JensenLab Experiment RNA-seq</a></td>

            <td><a href="/idg/targets?facet=Data+Source/JensenLab+Experiment+RNA-seq">15916</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/JensenLab+Experiment+RNA-seq"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/JensenLab+Experiment+RNA-seq"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://tissues.jensenlab.org">JensenLab Experiment UniGene</a></td>

            <td><a href="/idg/targets?facet=Data+Source/JensenLab+Experiment+UniGene">14935</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/JensenLab+Experiment+UniGene"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/JensenLab+Experiment+UniGene"></a>
            </td>
          </tr>

          <tr>


            <td>JensenLab Knowledge GHR</td>


            <td><a href="/idg/targets?facet=Data+Source/JensenLab+Knowledge+GHR">2052</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/JensenLab+Knowledge+GHR">705</a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/JensenLab+Knowledge+GHR"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://diseases.jensenlab.org">JensenLab Knowledge UniProtKB-KW</a></td>

            <td><a href="/idg/targets?facet=Data+Source/JensenLab+Knowledge+UniProtKB-KW">2279</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/JensenLab+Knowledge+UniProtKB-KW">119</a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/JensenLab+Knowledge+UniProtKB-KW"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://tissues.jensenlab.org">JensenLab Knowledge UniProtKB-RC</a></td>

            <td><a href="/idg/targets?facet=Data+Source/JensenLab+Knowledge+UniProtKB-RC">16759</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/JensenLab+Knowledge+UniProtKB-RC"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/JensenLab+Knowledge+UniProtKB-RC"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://diseases.jensenlab.org">JensenLab Text Mining</a></td>

            <td><a href="/idg/targets?facet=Data+Source/JensenLab+Text+Mining">15169</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/JensenLab+Text+Mining">4108</a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/JensenLab+Text+Mining"></a>
            </td>
          </tr>

          <tr>


            <td><a href="http://www.genome.jp/kegg/drug/">KEGG DRUG</a></td>


            <td><a href="/idg/targets?facet=Data+Source/KEGG+DRUG"></a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/KEGG+DRUG"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/KEGG+DRUG">44</a>
            </td>
          </tr>

          <tr>


            <td><a href="https://monarchinitiative.org/">Monarch</a></td>


            <td><a href="/idg/targets?facet=Data+Source/Monarch">3826</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/Monarch">5122</a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/Monarch"></a>
            </td>
          </tr>

          <tr>


            <td><a href="https://omabrowser.org">OMA</a></td>


            <td><a href="/idg/targets?facet=Data+Source/OMA">17786</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/OMA"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/OMA"></a>
            </td>
          </tr>

          <tr>

            <td><a href="http://omim.org/">OMIM</a></td>

            <td><a href="/idg/targets?facet=Data+Source/OMIM">3731</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/OMIM"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/OMIM"></a>
            </td>
          </tr>

          <tr>


            <td>Reactome</td>


            <td><a href="/idg/targets?facet=Data+Source/Reactome">1920</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/Reactome"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/Reactome"></a>
            </td>
          </tr>

          <tr>


            <td>SCIENTIFIC LITERATURE</td>


            <td><a href="/idg/targets?facet=Data+Source/SCIENTIFIC+LITERATURE"></a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/SCIENTIFIC+LITERATURE"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/SCIENTIFIC+LITERATURE">251</a>
            </td>
          </tr>

          <tr>

            <td><a href="http://habanero.health.unm.edu">TCRDv5.2.0</a></td>

            <td><a href="/idg/targets?facet=Data+Source/TCRDv5.2.0">20244</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/TCRDv5.2.0">19880</a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/TCRDv5.2.0">221871</a>
            </td>
          </tr>

          <tr>

            <td><a href="http://www.uniprot.org">UniProt</a></td>

            <td><a href="/idg/targets?facet=Data+Source/UniProt">20244</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/UniProt"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/UniProt"></a>
            </td>
          </tr>

          <tr>

            <td><a href="">UniProt Tissue</a></td>

            <td><a href="/idg/targets?facet=Data+Source/UniProt+Tissue">17989</a>
            </td>
            <td><a href="/idg/diseases?facet=Data+Source/UniProt+Tissue"></a>
            </td>
            <td><a href="/idg/ligands?facet=Data+Source/UniProt+Tissue"></a>
            </td>
          </tr>

          <tr>
            <td><a href="http://amp.pharm.mssm.edu/Harmonizome/">Harmonizome</a></td>
            <td>18912</td>
            <td></td>
            <td></td>
          </tr>
          </tbody></table>
        </div>
      </div>
    </div>
    <div fxLayout="row" id="datadownload" #datadownload #scrollSection>
      <div fxFlex="75">
        <div class="section-header" [ngClass]="{'active': isActive('datadownload')}">Data Download</div>
        <div class="section-content">
        <p>The current and past versions of TCRD are available from <a
          href="http://juniper.health.unm.edu/tcrd/download/">
          here</a>
        </p>
        </div>
      </div>
    </div>
    <div fxLayout="row" id="license" #license #scrollSection>
      <div fxFlex="75">
        <div class="section-header" [ngClass]="{'active': isActive('license')}">Pharos License</div>
        <div class="section-content">
        <p>All data accessed from Pharos is made available under the <a
          href="https://creativecommons.org/licenses/by-sa/4.0/">
          CC BY-SA 4.0</a> license.</p>
        </div>
      </div>
    </div>
    <div fxLayout="row" id="code" #code #scrollSection>
      <div fxFlex="75">
        <div class="section-header" [ngClass]="{'active': isActive('code')}">Pharos Code</div>
        <div class="section-content">
        <p>The sources for the Pharos web interface are available from <a href="https://spotlite.nih.gov/ncats/pharos">
          https://spotlite.nih.gov/ncats/pharos</a>. The repository provides
          <a href="https://spotlite.nih.gov/ncats/pharos/tree/master">instructions</a>
          on building and installation</p>
        </div>
      </div>
    </div>
    <div fxLayout="row" id="attribution" #attribution #scrollSection>
      <div fxFlex="75">
        <div class="section-header" [ngClass]="{'active': isActive('attribution')}">Attribution</div>
        <div class="section-content">
        <p>If you use Pharos, please consider citing it as
        <blockquote class="active blockquote">
          Nguyen, D.-T., Mathias, S. et al, "Pharos: Collating Protein Information to Shed Light on the Druggable
          Genome",
          <i>Nucl. Acids Res.</i>i>, <b>2017</b>, <i>45</i>(D1), D995-D1002.
          DOI: <a href="http://dx.doi.org/10.1093/nar/gkw1072">10.1093/nar/gkw1072</a>
        </blockquote>
      </div>
      </div>
    </div>
    <div fxLayout="row" id="help" #help #scrollSection>
      <div fxFlex="75">
        <div class="section-header" [ngClass]="{'active': isActive('help')}">Help</div>
        <div class="section-content">
        <p>for help in Pharos, click the help icon <mat-icon class="help-icon">help</mat-icon> or check the
          <a routerLink="/faq">Faq</a> page</p>
        <p>For feedback and comments, please contact us at <a href = "mailto:pharos@mail.nih.gov?subject=Pharos feedback">pharos@mail.nih.gov</a></p>
          <div class="slack-icon-span"><a href = "https://druggablegenome.slack.com" target="_blank">Join our slack channel<img src ="./assets/images/slack-icon.png" alt ="slack web icon link" class="slack-icon"></a></div>
        </div>
      </div>
    </div>
    <div class="spacer"></div>
  </div>
</div>

